lib/commands/context.js

```diff
@@ -61,8 +65,11 @@ extensions.isWebview = function isWebview () {
 };
 
 extensions.getNewRemoteDebugger = async function getNewRemoteDebugger () {
-  const socketPath = await this.opts.device.getWebInspectorSocket();
-  return new RemoteDebugger({
+  let socketPath;
+  if (!this.isRealDevice()) {
+    socketPath = await this.opts.device.getWebInspectorSocket();
+  }
+  return createRemoteDebugger({
```

Comment:
is it fine to leave socketPath undefined?

Reply:
Yes. `appium-remote-debugger` handles that (on real devices, it is ignored, on sims it depends on the version).